File: helpdeskangular/src/app/entity/TicketResponse.ts
export class TicketResponse {

    constructor(

        // ticket id
        public ticketid: number,
        // subject
        public subject: string,
        // creator name
        public creatorName: string,
        // assignee name
        public assigneeName: string,
        // create date time
        public createDatetime: Date,
        // ticket status name
        public ticketStatusName: string,
        // sla(service level agreement): ontime or late
        public sla: string,
        // last update date time
        public lastUpdateDatetime: Date,
        // spend hours. ex: 1.5 hours
        public spendHour: number,
        // spend hours in hh:mm:ss format. ex: 01:30:00
        public spendHourHhmmss: string,
        // creator phone number
        public creatorPhone: string,
        // creator email
        public creatorEmail: string,
        // team name
        public teamName: string,
        // category name
        public categoryName: string,
        // priority name
        public priorityName: string,

        // ticket content
        public content: string,

        // resolveIn
        public resolveIn: string,
        // currentDatetime
        public currentDatetime: Date,
        // ticket status id
        public ticketStatusid: number

    ) {

    }

}

File: helpdeskspringboot/src/main/java/com/ez/dto/TicketResponse.java
package com.ez.dto;

import java.util.Date;

public interface TicketResponse {

    // "ticketid" column
    Long getTicketid();

    // "subject" column
    String getSubject();

    // "creatorName" column
    String getCreatorName();

    // "assigneeName" column
    String getAssigneeName();

    // "createDatetime" column
    Date getCreateDatetime();

    // "ticketStatusName" column
    String getTicketStatusName();

    // "sla" column
    String getSla();

    // "lastUpdateDatetime" column
    Date getLastUpdateDatetime();

    // "spendHour" column
    // ex: 1.5 hours
    float getSpendHour();

    // "spendHourHhmmss" column
    // hours in hh:mm:ss(ex: 01:30:00)
    String getSpendHourHhmmss();

    // "creatorPhone" column
    String getCreatorPhone();

    // "creatorEmail" column
    String getCreatorEmail();

    // "teamName" column
    String getTeamName();

    // "categoryName" column
    String getCategoryName();

    // "priorityName" column
    String getPriorityName();

    // "content" column
    String getContent();

    // "resolveIn" column.
    // limit hours to resolve a ticket
    String getResolveIn();

    // "currentDatetime" column
    Date getCurrentDatetime();

    // "ticketStatusid" column
    long getTicketStatusid();

}
